kafo_parsers.gemspec

```diff
@@ -16,13 +16,15 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
+  spec.required_ruby_version = '>= 2.7', '< 4'
+
   spec.add_development_dependency "bundler", ">= 1.5", "< 3"
-  spec.add_development_dependency "rake"
-  spec.add_development_dependency "minitest"
-  spec.add_development_dependency "simplecov"
-  spec.add_development_dependency "ci_reporter_minitest"
+  spec.add_development_dependency "rake", "~> 13.0"
+  spec.add_development_dependency "minitest", "~> 5.0"
+  spec.add_development_dependency "simplecov", "~> 0.21"
+  spec.add_development_dependency "ci_reporter_minitest", "~> 1.0"
 
   # puppet manifests parsing
-  spec.add_dependency 'rdoc', '>= 3.9.0'
-  spec.add_dependency 'json'
+  spec.add_dependency "rdoc", ">= 3.9.0", "< 7"
+  spec.add_dependency "json", "~> 2.6"
```

Comment:
json is a bit special. It's actually shipped as part of Ruby and version 2.6 is quite aggressive. Here I'd use `~> 2.0` because Ruby 2.7 bundles json 2.3.